apm-sniffer/apm-sdk-plugin/tomcat-10x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat10x/TomcatInvokeInterceptor.java

```diff
@@ -82,9 +75,9 @@ public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allA
         HttpServletResponse response = (HttpServletResponse) allArguments[1];
 
         AbstractSpan span = ContextManager.activeSpan();
-        if (IS_SERVLET_GET_STATUS_METHOD_EXIST && response.getStatus() >= 400) {
+        Tags.HTTP_RESPONSE_STATUS_CODE.set(span, response.getStatus());
+        if (response.getStatus() >= 400) {
```

Comment:
I think this fixes the issue I mentioned

Reply:
> I think this fixes the issue I mentioned

But in the `EntrySpan`,

```java
    @Override
    public EntrySpan tag(String key, String value) {
        if (stackDepth == currentMaxDepth || isInAsyncMode) {
            super.tag(key, value);
        }
        return this;
    }
```

I suppose in the Tomcat plugin, the condition cannot be true. Do I miss something?

Reply:
It is better to debug this logic.

Reply:
![image](https://user-images.githubusercontent.com/2568208/222691111-56d8dee0-6c58-4ca6-b908-b961ac16cac1.png)

After debugging, it seems that the above method is not called. Instead, `public AbstractTracingSpan tag(AbstractTag<?> tag, String value)` is called while `EntrySpan` does not override this method (But `ExitSpan` does!). Then the `http.status_code` tag is set.

I am not sure if it is a feature or a bug? @wu-sheng @kezhenxu94

Reply:
Another question is why Tomcat and Spring set different status code...